src/asciidoc/docs/develop/testingTemplates.adoc

```diff
@@ -0,0 +1,208 @@
+
+Typically, you begin testing in the Koji editor.
+Later, you'll want to publish the template as *Unlisted*, so you can test it in a production environment before giving access to general users.
+
+== Testing in the Koji editor
+
+The Koji editor provides various ways to preview a Koji's behavior.
+
+=== Live preview
+
+The *Live preview* tab allows you to toggle between *Preview* mode, which shows what the Koji looks like when running, and *Editing* mode, which allows you to perform remixing in the preview.
+
+*Live preview* also allows you to preview the Koji in different contexts.
+You can display different template experiences depending on the context in which the template is running.
+For more on contexts, see <<withkoji-vcc-package#_context>>.
+
+The editor provides built-in previews for standard contexts in the *Live preview* pane.
+Some standard contexts, such as *Admin*, *About*, and *Sticker*, only appear if the developer has enabled them.
+
+To enable a standard context, create a page for the context, then create `.koji/project/entitlements.json`, if it does not already exist.
+Add an entry for the context in the file.
+For example, to enable an *Admin* page, add `"AdminContext": true` to `entitlements.json`.
+The following code, shows the contents of a simple `entitlements.json` file with an entry for an *Admin* page.
+
+[source,json]
+----
+{
+  "entitlements": {
+    "AdminContext": true
+  }
+}
+----
+
+Other standard contexts, such as <<withkoji-koji-iap-package#,in-app purchases>>, only appear if the developer has implemented them.
+The following code shows the contents of an `entitlements.json` file with in-app purchases enabled.
+It also shows <<withkoji-koji-iap-package#_iap,dynamic receipts>> enabled inside in-app purchases.
+
+[source,json]
+----
+{
+  "entitlements": {
+    "InAppPurchases": {
+      "enabled": true,
+      "products": [
+        {
+          "name": "Demo",
+          "sku": "demo",
+          "price": 100,
+          "dynamicReceipt": true
+        }
+      ]
+    },
+    "FeedEvents": false
+  }
+}
+----
+
+If you define a custom context in your application code, you can preview it in the *Live player* by appending the parameter to the URL. For example, `?context=custom`. You cannot preview a custom context in the *Live preview*.
+
+=== Remote previews
+
+The *Remote* tab offers several additional ways of previewing your Koji.
+
+* On a mobile device – Open a preview on any device by scanning a QR code.
+Logs from the device appear in a pane in the editor.
+
+* In a new tab – Open a preview on a staging server.
+Logs from the tab appear in a pane in the editor.
+
+* In the *Live player* – Open another browser window where the Koji is displayed in an iframe, just as the published Koji will display.
+You can also use the *Live player* to test <<_testing_in_app_purchases,in-app purchases>>.
```

Comment:
```suggestion
You can also use the *Live player* to <<_testing_in_app_purchases,test in-app purchases>>.
```
Minor thing, but I recommend including "test" in the contextual link text to distinguish it from a link to "in-app purchases" documentation. It gives readers a better sense of what they'll find if they follow the link.